plotly/tests/test_core/test_jupyter/lib/server.js

```diff
@@ -0,0 +1,93 @@
+var http = require('http');
+var url = require('url');
+var fs = require('fs');
+var path = require('path');
+var ecstatic = require('ecstatic');
+var browserify = require('browserify');
+var cheerio = require('cheerio');
+var chrome = require('chrome-launch');
+var tapParser = require('tap-parser');
+
+var PORT = 8080;
+var PATH_ROOT = path.join(__dirname, '..');
+var PATH_FIXTURES = path.join(PATH_ROOT, 'fixtures');
+var PATH_INDEX = path.join(PATH_FIXTURES, 'test.html');
+var PATH_INDEX_STUB = path.join(PATH_FIXTURES, 'test.tmp.html');
+var PATH_TEST_FILE = path.join(PATH_ROOT, 'test.js');
+var PATH_TEST_BUNDLE = path.join(PATH_ROOT, 'test.tmp.js');
+var URL = 'http://localhost:' + PORT + '/fixtures/test.tmp.html';
+var EXIT_CODE = 0;
+
+// main
+stubIndex()
+    .then(bundleTests)
+    .then(startServer)
+    .then(launch)
+
+function stubIndex() {
+    return new Promise(function(resolve, reject) {
+        var html = fs.readFileSync(PATH_INDEX, 'utf-8');
+        var $ = cheerio.load(html);
+
+        $('body').append('<script type="text/javascript" src="../test.tmp.js"></script>');
```

Comment:
Yeah. Maybe this step a bit too magical. Ideally, I want to keep the `fixtures/*.ipynb` as close to real NBs as possible. I guess we could add this `<script>` after converting `ipynb` -> `html` in the `_test.py` file. But I thought that it made more sense to have the test runner add the test `<script>` ... to maybe one release this as a standalone package :smile: